### PR TITLE
Cleanup external data deps

### PR DESCRIPTION
**Pull request by trey-stafford**
This PR focuses on cleaning up external data dependencies for `pm_icecon`.

This includes:
* Removal of code related to reading masks from NSDIC-specific NFS locations
* Removal of CLI for `pm_icecon`. 

This is a step in the direction of making `pm_icecon` purely responsible for defining the sea ice concentration algorithms. Projects like `seaice_ecdr` will be responsible for providing input TBs, invalid ice masks, land mask, etc.

Related PR for `seaice_ecdr`: https://github.com/nsidc/seaice_ecdr/pull/57

### CODE REVIEW

File: pm_icecon/bt/api.py
Comment (trey-stafford) [inline pm_icecon/bt/api.py:1]: This and the `nt` API still fetch external (currently requiring NSIDC NFS access) data via `pm_tb_data`.

Eventually we could cleanup these APIs that fetch data and instead make the focus a clear and clean interface to the main nt/bt algorithms with descriptions for all required inputs. A user would then e.g., import something like `nasateam_seaice_conc()`, which would require that the user pass in TBs, masks, etc. required for running that algorithm and getting an SIC estimate as a result.

File: pm_icecon/tests/regression/test_bt.py
Comment (trey-stafford) [inline pm_icecon/tests/regression/test_bt.py:1]: Still need to organize regression data in a new area. Will address that separately. For now, I moved a lot of the data/ancillary info accessing code to the tests where they're used.

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "pm_icecon"
-version = "0.2.0"
+version = "0.3.0"
```
Comment (trey-stafford) [inline pyproject.toml:3]: Until we release v1.0.0, minor version changes signal breaking changes. This is definitely a breaking-change PR!

Comment (sc0tts) [reply pyproject.toml:3]: Agree!

**Review by sc0tts — APPROVED** [review]
This all looks sooo much cleaner.  Very good!

**Pull request merged**